yjit/src/core.rs

```diff
@@ -1472,17 +1481,15 @@ fn gen_block_series_body(
         let mut last_branch = last_branchref.borrow_mut();
 
         // gen_direct_jump() can request a block to be placed immediately after by
-        // leaving `None`s in the `dst_addrs` array.
-        match &last_branch.dst_addrs {
-            [None, None] => (),
-            _ => {
-                break;
-            } // If there is no next block to compile, stop
+        // leaving a single target that has a `None` address.
```

Comment:
Off topic, while making this patch I noticed that this use case is really the only thing stopping us from making `BrachTarget::address` non-optional. With that change, we can make `Block::start_addr` non-optional too, which would be a nice improvement.

Reply:
That would be a nice improvement